sdk/storage/Azure.Storage.DataMovement/src/JobPlan/JobPlanHeader.cs

```diff
@@ -5,6 +5,7 @@
 using System.IO;
 using System.Text;
 using Azure.Core;
+using static Azure.Storage.DataMovement.CheckpointerExtensions;
```

Comment:
Feels weird to be statically using an "extensions" class. Looking at the impl, it seems the BinaryWriter isn't a `this` parameter. Consider making it one so you can remove this `using static` and have the below changes read `writer.WriteVariableLengthFieldinfo(parentSourcePathBytes.Length, ref currentVariableLengthIndex);`. This will follow extension expectations.

Reply:
Good point, will do!